core/shared/src/main/scala/coursier/util/Parse.scala

```diff
@@ -191,59 +206,81 @@ object Parse {
         module(s"$org::$rawName", defaultScalaVersion)
           .right
           .map(mod => {
-            Dependency(
-              mod,
-              version,
-              config,
-              attributes,
-              transitive = transitive,
-              exclusions = localExcludes.getOrElse(mod.orgName, Set()) | globalExcludes)
+            (Dependency(
+                mod,
+                version,
+                config,
+                attributes,
+                transitive = transitive,
+                exclusions = localExcludes.getOrElse(mod.orgName, Set()) | globalExcludes),
+              extraDependencyParams)
```

Comment:
As `extraDependencyParams` is added to all the right cases, we could do something like
```scala
val depOrError = parts match {
   …
}

depOrError.right.map(dep => (dep, extraDependencyParams))
```
(only adding extraDependencyParams at one place).